src/Groups/matrices/form_group.jl

```diff
@@ -184,7 +184,7 @@ An exception is thrown if `base_ring(G)` is not a finite field with even degree
 over its prime subfield.
 
 !!! warning "Note:"
-    At the moment, elements of the generating set are returned of type `mat_elem_type(G)`.
+    At the moment, elements of the generating set are returned of type `mat_elem_type(typeof(G))`.
```

Comment:
For other similar `*_type` function we actually make sure that one can also pass in instances, for convenience. Maybe a good idea to do that here, too? E.g. we do this in AA:
```julia
elem_type(x) = elem_type(typeof(x))
elem_type(T::DataType) = throw(MethodError(elem_type, (T,)))
```

Reply:
I found `mat_elem_type` a bit irritating: It is not documented at all, but it appears essentially in these warnings in the documentation.

Now I think that the function is useful. Thus I will document `mat_elem_type` and `ring_elem_type`, and then it makes sense to admit also the group as an argument instead of its type.